hack/update: kubernetes version

prezha opened the pull request:
fixes: #9464
fixes: #4392
replaces: #9384 

pr #9464 ("automate update processes") split: part 2/4:
- includes: `hack/update/kubernetes_version/update_kubernetes_version.go`
- removes: `hack/kubernetes_version/update_kubernetes_version.go`

<!-- 🎉 Thank you for contributing to minikube! 🎉 Here are some hints to get your PR merged faster:

1. Your PR title will be included in the release notes, choose it carefully
2. If the PR fixes an issue, add "fixes #<issue number>" to the description.
3. If the PR is a user interface change, please include a "before" and "after" example.
4. If the PR is a large design change, please include an enhancement proposal:
https://github.com/kubernetes/minikube/tree/master/enhancements
-->

tstromberg reviewed (changes requested):
Looks good, but references `hack/update` directory which doesn't yet exist. We'll need to defer merging until that directory is filled in. 

Please resolve this comment once that code is merged.

tstromberg commented:
Dependency updated - can you resolve the conflicts?

prezha commented:
@tstromberg similar as with pr #9532:

the conflicts were due to the pr #9464 (the original one - before i've split it into several prs) got merged first, and i had a couple of changes made in those individual prs; essentially my fault, i should have had closed the original pr right after the split (but left it just in case anyone wanted to compare)...

i've fixed this one now but had to add `base library` from pr #9526 to resolve dependencies, should be ok to merge this as well as pr #9526 (i'll resolve the conflicts there if any)
